File: snap_prep/0001.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numSet = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int n1 = nums[i];
            int n2 = target - n1;
            if(numSet.containsKey(n2) == true){
                int j = numSet.get(n2);
                int[] ans = {i,j};
                return ans;
            }
            numSet.put(n1, i);
        }
        return null;
    }
}

File: snap_prep/0001.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> null;
        map<int, int> num_set;
        for(int i = 0; i < nums.size(); i++){
            int n1 = nums[i];
            int n2 = target - n1;
            auto j = num_set.find(n2);
            if(j != num_set.end()){
                vector<int> ans{i, j->second};
                return ans;
            }
            num_set.insert(pair<int, int>(n1, i));
        }
        return null;
    }
};
